public/javascripts/script.js

```diff
@@ -1,174 +1,77 @@
+            }).on('dragstart', function () {
+                $priorityEvents.addClass('droppable');
+            }).on('dragstop', function () {
+                $priorityEvents.removeClass('droppable');
+            });
+
+            $priorityEvents.sortable({
+                axis: 'y'
+            });
+
+            $priorityEvents.on('sortupdate', function (event, ui) {
+                eventOrdering.crunchJsonAndSend();
+            }).on('sortreceive', function (event, ui) {
+                $priorityEvents.removeClass('droppable');
+
+                if ($priorityEvents.find('.event').length < 5) {
+                    ui.item.remove();
+                    eventOrdering.bindCancel($priorityEvents);
+                } else {
+                    var newId = ui.item.data('id');
+                    $priorityEvents.find('[data-id=' + newId + ']').remove();
+                    alert('Only four priority events can be selected, please remove an old event if you wish to add a new one.');
+                }
+            });
+
+            eventOrdering.parseDates();
+            eventOrdering.bindCancel($priorityEvents);
+        }
+    },
+    crunchJsonAndSend: function () {
+        var $listItems = $('#eventsPriority > li');
+        var order = [];
 
-        }).fail(function fail () {
-          alert("Upload failed, please retry.");
-          self.$popover.popover("hide");
+        $listItems.each(function (i, e) {
+            order.push($(e).data('id'));
         });
 
-      } else {
-        self.$popover.popover("hide");
-      }
-
-    });
-
-  } else {
-    alert("Please choose a file to upload.");
-  }
-
-};
-
-var checkFile = function (file, callback) { // check an images aspect ratio using the HTML5 file api
-  var confirmMessage = [];
-  confirmMessage.push("This image is not at the correct 5:3 Aspect Ratio (eg: 1920x1152).");
-  confirmMessage.push("Uploading this image will probably break the layout on the membership pages or display in an unfavourable way.");
-  confirmMessage.push("Please confirm you wish to upload this image (you have been warned!)");
-  var reader = new FileReader();
-  reader.onload = function(e) {
-    try {
-      var img = new Image();
-      img.src = e.target.result;
-      var w = img.naturalWidth, h = img.naturalHeight, fiveByThree = 1920/1152;
-
-      if (w/h === fiveByThree) {
-        callback(true);
-      } else {
-        if (confirm(confirmMessage.join("\n\r"))) {
-          callback(true);
-        } else {
-          callback(false);
-        }
-      }
-
-    } catch (e) {
-      alert("The image file could not be read: " + e);
-      callback(false);
+        $.post('/order', {
+            order: order
+        }, function () {
+            $('.selector').draggable({ disabled: false });
+        });
+    },
+    bindCancel: function ($priorityEvents) {
+        $priorityEvents.find('.event .remove').off('click').on('click', function (event) { // DIRTY!
```

Comment:
In all honesty I'm not sure what this is doing @chrisfinch code thats been moved around. 
Bar name spacing this clientside JS I haven't done anything else. 
I feel if this is to be used more then this could all be re-written in vanilla JS with micro libs and remove JQuery. But at this point I stopped myself from going down a refactoring rabbit hole.